### ISSUE TITLE
addGlPolygons not rendering polygons when shapefile is too large?

### ISSUE THREAD

**Issue by bmaparicio** [r-spatial/leafgl#91]
Cheers for this incredibly helpful package!
When I was trying to use it in a large shapefile (ca. 150k polygons), the map did not show any of my polygons, just the base map. 
It works well when I use a smaller sample from the same shapefile (like 20k). I'm using the github version but had the same problem with the cran version.

Here is my script.

leaflet() %>%
  leaflet::addProviderTiles('Esri.NatGeoWorldMap', group = "Esri.NatGeoWorldMap") %>% 
  leafgl::addGlPolygons(data = my_data,
                color = "black",
                weight = 0.1,
                popup = "landuse",
                group = "landuse") %>%
  addLayersControl(overlayGroups = "landuse")

Am I doing something wrong or is there a limitation in the amount of polygons one can use?

**Comment by trafficonese** [comment]
Can you maybe check the browser console for errors aswell?
It's hard to say whats happening without your data..

**Comment by bmaparicio** [comment]
In the meantime, I simplified the polygons' shape and it shows. Maybe it is a memory problem? 
I can send you my data if you have time to figure out what is happening. 
Thanks!!

**Comment by tim-salabim** [comment]
Maybe some polygons are invalid?

**Comment by bmaparicio** [comment]
I'm making sure they are valid with st_make_valid before doing anything. Anyway, simplifying the shape worked for me and made everything much faster. Thanks!

**Comment by tim-salabim** [comment]
Id still be interested in what caused your problem in the first place, so if you could share your data that would be great

**Comment by bmaparicio** [comment]
Yes, for sure. Share your email contact and I'll send it. Thanks a lot!

tim-salabim ***@***.***> escreveu (segunda, 18/03/2024 à(s)
11:23):

**Comment by tim-salabim** [comment]
You can find my email in the `DESCRIPTION` file ;-)
